server/api/pprof.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"runtime/pprof"
+	"strconv"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/server"
+	"github.com/tikv/pd/server/versioninfo"
+	"github.com/unrolled/render"
+	"go.uber.org/zap"
+)
+
+// ProfHandler pprof handler
+type ProfHandler struct {
+	svr *server.Server
+	rd  *render.Render
+}
+
+// newProfHandler constructor for ProfHandler
+func newProfHandler(svr *server.Server, rd *render.Render) *ProfHandler {
+	return &ProfHandler{
+		svr: svr,
+		rd:  rd,
+	}
+
+}
+
+// @Summary debug zip of PD servers.
+// @Produce application/octet-stream
+// @Router /pprof/debug/zip [get]
+func (h *ProfHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Disposition", fmt.Sprintf(`attachment; filename="pd_debug"`+time.Now().Format("20060102_150405")+".zip"))
```

Comment:
The format is a little bit different with TiDB.

Reply:
yes, it will be better to splite date and time